### ISSUE TITLE
search_messages silently returns stale/incomplete results (process-global lexical bridge) → false-negatives

### ISSUE THREAD

**Issue by bluejay-creators** [Dicklesworthstone/mcp_agent_mail_rust#227]
## Summary

`search_messages` (and `am robot search`) can silently return **stale, incomplete results with full confidence**, causing callers to conclude a delivered message does not exist. Two related problems, the second more serious than the first.

Version: **am 0.3.24** (Rust). Multi-project setup: a fleet project (`/Users/jubin`) plus per-repo projects.

## Repro / concrete consequence

An agent searched its fleet project for the term `asyncEligible` to check whether a decision-relay message had arrived. The result set contained messages #732 and #848 but **not #733** — a delivered message addressed to it that contained the search term. It concluded (reasonably, and wrongly) that the relay never happened and reported that in a status update. The message existed the whole time; an identical query moments later from a different caller returned it.

## Diagnostics from the degraded query

```json
"degraded": true,
"fallback_mode": "search_index_stale",
"search_index_state": "stale",
"search_index_reason": "process-global lexical bridge is active for a different database: /Users/.../git_mailbox_repo/storage.sqlite3",
"remediation_hints": ["Run `am robot search <query>` to refresh Search V3 lexical backfill"]
```

## Issue 1 — process-global lexical bridge → stale cross-project search

The Search V3 lexical index/bridge appears to be **process-global: it serves one project's DB at a time**. In a multi-project deployment, a search against project A while the bridge is bound to project B returns a stale/incomplete index for A. Backfill is query-triggered and CWD/`--project`-scoped, so the "fix" is to re-run a search against the intended project — but any subsequent search against another project re-staleness it. There is no dedicated reindex command (`am robot` exposes only `navigate` and `search`); `am robot search --project <p>` is the only refresh path.

## Issue 2 (larger) — degraded search returns incomplete results silently

When degraded, the search still returns a normal-looking result set. The `degraded` / `fallback_mode` / `search_index_state` fields flag it, **but nothing forces a caller to read them**, and a missing hit is indistinguishable from a genuinely-absent message. For autonomous agents this is worse than an error: "not found" is silently conflated with "not indexed," and routing/escalation decisions get made on the false-negative.

**Suggested direction:** when the index is known-degraded/stale, either (a) surface it *in-band and loudly* (e.g. a top-level `warning`/`partial: true` the caller can't miss, or a non-zero status), or (b) refuse the query and return an explicit "index stale, refresh required" error rather than a silently-incomplete result set. A false-negative from a search consumed by automation should never look like a clean empty/partial result.

Happy to provide more detail. Thanks for the fast turnaround on the v0.3.24 read/reply fixes.

**Comment by Dicklesworthstone** [comment]
Fixed on main; here is the verified mechanism behind your false-negative, which is subtly different from (and worse than) the stale-bridge story the diagnostics told.

**What actually produced "confident stale results":** every message ingestion path funnels through `search_v3::index_message`, and the process-wide search **result cache** (300s TTL by default) was only invalidated when the message was successfully indexed into the Tantivy bridge. When the process-global bridge is uninitialized *or bound to a different database* — your multi-project deployment — deliveries skip lexical indexing (`Ok(false)`) and therefore **never invalidated the cache**. So a search cached at T0 (before #733's delivery) kept answering the identical query with the T0 result set for up to the TTL, with full confidence. A different caller has a different cache key (the key includes authorization context), which is exactly why "an identical query moments later from a different caller returned it."

**Fixes on main:**
1. `index_message` / `index_messages_batch` now invalidate the search cache **unconditionally** on ingestion — bridge present or not, foreign or not, Tantivy write failed or not (commit c7ca58d4; regression test `gh227_index_message_invalidates_search_cache_without_bridge`).
2. On your issue 1 (cross-project searches while the bridge is bound elsewhere): since GH#162 (shipped in v0.3.24), the `state == "stale"` signal routes lexical/hybrid retrieval to a SQL scan against *this pool's* relational tables, so results are complete for keyword queries even while degraded — the cache hole above was what made that fallback appear to return stale sets. The `degraded/search_index_stale` diagnostics remain (accurate: the lexical index genuinely isn't serving that pool), and the deeper design item — per-database lexical indexes instead of one process-global bridge — is real but is an architecture change; the correctness hole is closed with the two fixes above.

**Verify:** on a build past v0.3.24, run your fleet search, deliver a message containing the search term from another project's client, and repeat the identical query within the TTL window — the new message appears immediately.
